File: easy/c/0021.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* mergeTwoLists(struct ListNode* l1, struct ListNode* l2){
    if(l1 == NULL)return l2;
    if(l2 == NULL)return l1;
    struct ListNode* answer = malloc(sizeof(struct ListNode));
    struct ListNode* current = answer;
    while(l1!=NULL && l2!=NULL){
        if(l1->val >= l2->val){
            current->next = l2;
            l2 = l2->next;
        }else{
            current->next = l1;
            l1 = l1->next;
        }
        current = current->next;
        if(l1 == NULL)current->next = l2;
        if(l2 == NULL)current->next = l1;
    }
    return answer->next;
}

File: easy/go/0021.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(l1 *ListNode, l2 *ListNode) *ListNode {
    answer := &ListNode{}
    current := answer
    if l1 == nil{
        return l2
    }
    if l2 == nil{
        return l1
    }
    for l1 != nil && l2 != nil{
        if l1.Val >= l2.Val{
            current.Next = l2
            l2 = l2.Next
        }else{
            current.Next = l1
            l1 = l1.Next
        }
        current = current.Next
        if(l1 == nil){
            current.Next = l2
        }
        if(l2 == nil){
            current.Next = l1
        }
    }
    return answer.Next
}